### PR TITLE
use prettier to auto format md files

### LINKED ISSUE #421: use a consistent syntax format for md files
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
(This section helps Arrow developers understand the context and *why* for this feature, in addition to  the *what*)

use a consistent syntax format for md files so that we no longer need to manually format files or create unnecessary diffs.

**Describe the solution you'd like**

use a GitHub action that utilizes `prettier` from https://prettier.io

**Describe alternatives you've considered**

setup a pre-commit job but it requires manual setup for all developers

**Additional context**

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

Closes #421.

 # Rationale for this change

we can use `prettier` to auto-format `.md` or even `.yaml` files and set it as a GitHub action to auto-correct.

# What changes are included in this PR?

- added the GitHub action in `dev` workflow
- the second commit is done automatically and pushed by the GitHub action

# Are there any user-facing changes?

no user facing changes as prettier would not change semantics of docs

### CODE REVIEW

**Comment by Jimexist** [comment]
Now it's automatic! Neat!

**Review by msathis — APPROVED** [review]
Great work @Jimexist 👍

**Review by alamb — COMMENTED** [review]
Thank you very much @Jimexist  -- this would be awesome. I don't think we can use the third party action on this repo, but I think we could bring the code in directly if you wanted to try.

👍

**Comment by alamb** [comment]
FYI @Dandandan

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -23,7 +23,22 @@ on:
...
 
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Make sure the actual branch is checked out when running on pull requests
+          ref: ${{ github.head_ref }}
+
+      - name: Prettify code
+        uses: creyD/prettier_action@v3.3
```
[the remark this answers is not in the document]
Comment (Jimexist) [reply .github/workflows/dev.yml:37]: good point. do you think copying it to be something under apache org repo is desired?

[the remark this answers is not in the document]
Comment (alamb) [reply .github/workflows/dev.yml:37]: > good point. do you think copying it to be something under apache org repo is desired?

Yes, I think that would be preferable (the rationale being that then that someone can't change the contents of the action in the third party repository and affect the apache org via a github action)

**Comment by alamb** [comment]
Seems like this is supserceded by https://github.com/apache/arrow-datafusion/pull/453 so closing this to keep the PR queue down. Please reopen it @Jimexist  if that is not correct

**Pull request closed without merging**